server/irmaserver/helpers.go

```diff
@@ -48,15 +48,42 @@ func (session *session) onUpdate() {
 	session.sse.SendMessage("session/"+session.clientToken,
 		sse.SimpleMessage(fmt.Sprintf(`"%s"`, session.status)),
 	)
-	session.sse.SendMessage("session/"+session.token,
+	session.sse.SendMessage("session/"+session.backendToken,
 		sse.SimpleMessage(fmt.Sprintf(`"%s"`, session.status)),
 	)
 }
 
+// Checks whether requested options are valid in the current session context.
+func (session *session) updateFrontendOptions(request *irma.OptionsRequest) (*server.SessionOptions, error) {
+	if session.status == server.StatusInitialized {
+		if request.BindingMethod == irma.BindingMethodNone {
+			session.options.BindingMethod = irma.BindingMethodNone
+			session.options.BindingCode = ""
+		} else if request.BindingMethod == irma.BindingMethodPin {
+			session.options.BindingMethod = irma.BindingMethodPin
```

Comment:
Somehow I have to check that request did not include a binding method that we don't know. Otherwise the IRMA app will receive a method which it cannot render. I can do something like:
```
if request.BindingMethod != irma.BindingMethodNone && request.BindingMethod != irma.BindingMethodPin {
  return nil, errors.New("Binding method unknown")
}
session.options.BindingMethod = request.BindingMethod
if request.BindingMethod == irma.BindingMethodPin {
  session.options.BindingCode = common.NewBindingCode()
} else {
  session.options.BindingCode = ""
}
```

It is however more lines of code than I previously had.

Reply:
I now did it somewhat different by moving the `BindingMethod` assignment to the end. This should make it better.